CMakeLists.txt

```
@@ -2385,144 +2385,187 @@ endif()
-  #src/test/effectchainslottest.cpp
-  src/test/enginebufferscalelineartest.cpp
-  src/test/enginebuffertest.cpp
-  src/test/engineeffectsdelay_test.cpp
-  src/test/enginefilterbiquadtest.cpp
-  src/test/enginemixertest.cpp
-  src/test/enginemicrophonetest.cpp
-  src/test/enginesynctest.cpp
-  src/test/fileinfo_test.cpp
-  src/test/frametest.cpp
-  src/test/globaltrackcache_test.cpp
-  src/test/hotcuecontrol_test.cpp
-  src/test/imageutils_test.cpp
-  src/test/indexrange_test.cpp
-  src/test/itunesxmlimportertest.cpp
-  src/test/keyfactorytest.cpp
-  src/test/keyutilstest.cpp
-  src/test/lcstest.cpp
-  src/test/learningutilstest.cpp
-  src/test/libraryscannertest.cpp
-  src/test/librarytest.cpp
-  src/test/looping_control_test.cpp
-  src/test/main.cpp
-  src/test/mathutiltest.cpp
-  src/test/metadatatest.cpp
-  #TODO: make this build again
-  #src/test/metaknob_link_test.cpp
-  src/test/midicontrollertest.cpp
-  src/test/mixxxtest.cpp
-  src/test/mock_networkaccessmanager.cpp
-  src/test/movinginterquartilemean_test.cpp
-  src/test/musicbrainzrecordingstasktest.cpp
-  src/test/nativeeffects_test.cpp
-  src/test/performancetimer_test.cpp
-  src/test/playcountertest.cpp
-  src/test/playermanagertest.cpp
-  src/test/playlisttest.cpp
-  src/test/portmidicontroller_test.cpp
-  src/test/portmidienumeratortest.cpp
-  src/test/queryutiltest.cpp
-  src/test/rangelist_test.cpp
-  src/test/readaheadmanager_test.cpp
-  src/test/replaygaintest.cpp
-  src/test/rescalertest.cpp
-  src/test/rgbcolor_test.cpp
-  src/test/rotary_test.cpp
-  src/test/ringdelaybuffer_test.cpp
-  src/test/samplebuffertest.cpp
-  src/test/sampleutiltest.cpp
-  src/test/schemamanager_test.cpp
-  src/test/searchqueryparsertest.cpp
-  src/test/seratobeatgridtest.cpp
-  src/test/seratomarkerstest.cpp
-  src/test/seratomarkers2test.cpp
-  src/test/seratotagstest.cpp
-  src/test/signalpathtest.cpp
-  src/test/skincontext_test.cpp
-  src/test/softtakeover_test.cpp
-  src/test/soundproxy_test.cpp
-  src/test/soundsourceproviderregistrytest.cpp
-  src/test/sqliteliketest.cpp
-  src/test/synccontroltest.cpp
-  src/test/synctrackmetadatatest.cpp
-  src/test/tableview_test.cpp
-  src/test/taglibtest.cpp
-  src/test/trackdao_test.cpp
-  src/test/trackexport_test.cpp
-  src/test/trackmetadata_test.cpp
-  src/test/trackmetadataexport_test.cpp
-  src/test/tracknumberstest.cpp
-  src/test/trackreftest.cpp
-  src/test/trackupdate_test.cpp
-  src/test/uuid_test.cpp
-  src/test/wbatterytest.cpp
-  src/test/wpushbutton_test.cpp
-  src/test/wwidgetstack_test.cpp
-  src/test/waveform_upgrade_test.cpp
-  src/util/moc_included_test.cpp
-  src/test/helpers/log_test.cpp
-)
-if(QML)
-  target_sources(
+if(NOT DEFINED BUILD_TESTING)
+  find_package(GTest CONFIG)
+  if(GTest_FOUND)
+    message(STATUS "Found GTest: tests enabled")
+    set(BUILD_TESTING ON)
+  endif()
+elseif(BUILD_TESTING)
+  find_package(GTest CONFIG REQUIRED)
+  if(GTest_FOUND)
+    message(STATUS "Found GTest")
+  endif()
+endif()
```

What is the reason for not doing:

```suggestion
find_package(GTest CONFIG)
cmake_dependent_option(
  BUILD_TESTING
  "Build with Unittests"
  OFF
  "GTest_FOUND"
  ON
)
if(BUILD_TESTING AND NOT GTest_FOUND)
  message(FATAL_ERROR "GTest is required for DBUILD_TESTING=ON")
endif()
````

same below

---

None. But you mean ON/OFF reverted no ?

---

Oh yes sure. Sorry.

---

Oh, just noticed that cmake_dependent_option() does not work, because it forces the value to "OFF" independent form the user value. We want:  
`default_option(BUILD_TESTING, "Build with Unittests", "GTest_FOUND")`

---

The commit's logic is clear and CMakeMagic doesn't decrease the number of lines.

---

default_option(BUILD_TESTING, "Build with Unittests", "GTest_FOUND")

Reuses a bit of the complexity used elsewhere in our CMakLists.txt.
That's why I prefere this. But I don't insist.

---

OK. I'll check that.
PR 2.5 pushed.